packages/web-app-files/src/views/trash/Overview.vue

```diff
@@ -0,0 +1,232 @@
+            </div>
+          </template>
+        </oc-table>
+      </template>
+    </files-view-wrapper>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, nextTick, onMounted, ref, unref, watch } from 'vue'
+import Mark from 'mark.js'
+import Fuse from 'fuse.js'
+import { useGettext } from 'vue3-gettext'
+import { useTask } from 'vue-concurrency'
+import { useGraphClient, useRouter, useStore } from 'web-pkg'
+import { createLocationTrash } from 'web-app-files/src/router'
+import { createFileRouteOptions } from 'web-pkg/src/helpers/router'
+import AppBar from 'web-app-files/src/components/AppBar/AppBar.vue'
+import FilesViewWrapper from 'web-app-files/src/components/FilesViewWrapper.vue'
+import { isPersonalSpaceResource, isProjectSpaceResource } from 'web-client/src/helpers'
+import AppLoadingSpinner from 'web-pkg/src/components/AppLoadingSpinner.vue'
+
+export default defineComponent({
+  name: 'TrashOverview',
+  components: { FilesViewWrapper, AppBar, AppLoadingSpinner },
+  setup: function () {
+    const store = useStore()
+    const router = useRouter()
+    const { $gettext } = useGettext()
+    const sortBy = ref('name')
+    const sortDir = ref('asc')
+    const filterTerm = ref('')
+    const markInstance = ref(undefined)
+    const tableRef = ref(undefined)
+    const { graphClient } = useGraphClient()
+
+    const spaces = computed(() => {
+      return orderBy(
+        store.getters['runtime/spaces/spaces'].filter(
+          (s) => isPersonalSpaceResource(s) || isProjectSpaceResource(s)
+        ),
+        unref(sortBy),
+        unref(sortDir)
+      )
+    })
+
+    const loadResourcesTask = useTask(function* () {
+      store.commit('Files/CLEAR_FILES_SEARCHED')
+      store.commit('Files/CLEAR_CURRENT_FILES_LIST')
+      yield store.dispatch('runtime/spaces/reloadProjectSpaces', {
+        graphClient: unref(graphClient)
+      })
+      store.commit('Files/LOAD_FILES', { currentFolder: null, files: unref(spaces) })
+    })
+
+    const footerTextTotal = computed(() => {
+      return $gettext('%{spaceCount} trashes in total', {
+        spaceCount: unref(spaces).length.toString()
+      })
+    })
+    const footerTextFilter = computed(() => {
+      return $gettext('%{spaceCount} matching trashes', {
+        spaceCount: unref(orderedSpaces).length
+      })
+    })
+
+    const breadcrumbs = computed(() => [
+      { text: $gettext('Deleted files'), onClick: () => loadResourcesTask.perform() }
+    ])
+
+    const orderBy = (list, prop, desc) => {
+      return [...list].sort((s1, s2) => {
+        if (s1.driveType === 'personal') {
+          return -1
+        }
+        if (s2.driveType === 'personal') {
+          return +1
+        }
+
+        const a = s1[prop]
+        const b = s2[prop]
+
+        return desc ? b.localeCompare(a) : a.localeCompare(b)
+      })
+    }
+    const orderedSpaces = computed(() =>
```

Comment:
Ah, dang. There is some mixup of `spaces` and `orderedSpaces` now. I think it makes sense like you had it before: first doing the filtering and afterwards the sorting (because that's much easier on the CPU).

Means: please remove the sorting from the `spaces` computed prop and instead do it here as it was before.

I think `orderedSpaces` is misleading, var name wise. `filteredSpaces` is IMO better, but still not good. `filteredOrderedSpaces` is just horrible. Something like `displaySpaces` would also be okish...

Reply:
Could you please commit it, I am a little confused

Reply:
You already committed the right changes 👍